### PR TITLE
indexes section in the HTML repr

### PR DESCRIPTION
**Pull request by keewis**
To see the effect, try this:

<details>

```python
import xarray as xr
from xarray.core.indexes import Index


class CustomIndex(Index):
    def __init__(self, names, options):
        self.names = names
        self.options = options
    
    @classmethod
    def from_variables(cls, variables, options):
        names = list(variables.keys())
        
        return cls(names, options)
    
    def __repr__(self):
        options = (
            {"names": repr(self.names)}
            | {str(k): str(v) for k, v in self.options.items()}
        )
        
        return f"CustomIndex({', '.join(k + '=' + v for k, v in options.items())})"

    def _repr_html_(self):
        header_row = "<tr><td>KDTree params</td></tr>"
        option_rows = [
            f"<tr><td>{option}</td><td>{value}</td></tr>"
            for option, value in self.options.items()
        ]
        return f"<left><table>{header_row}{''.join(option_rows)}</table></left>"

ds = xr.tutorial.open_dataset("rasm")
ds1 = ds.set_xindex(["xc", "yc"], CustomIndex, param1="a", param2="b")

with xr.set_options(display_style="text"):
    display(ds1)

with xr.set_options(display_style="html"):
    display(ds1)
```

</details>

~The repr looks a bit strange because I've been borrowing the variable CSS classes.~ Edit: @benbovy fixed that for me

Also, the discussion about what `_repr_inline_` should include from #7183 is relevant here as well.

- [x] Follow-up to #6795, depends on #7183
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by benbovy** [comment]
Thanks @keewis for opening this PR.

I added some commits (hope you don't mind) to fix the CSS. I also grouped the items in the indexes section by unique index with index coordinates separated by line return, so it looks like the coordinate section while the multi-coordinate indexes are clearly visible.

File: xarray/core/formatting_html.py [hunk 1]
```diff
@@ -213,6 +247,13 @@ def array_section(obj):
     expand_option_name="display_expand_data_vars",
 )
 
+index_section = partial(
+    _mapping_section,
+    name="Indexes",
+    details_func=summarize_indexes,
+    max_items_collapse=15,
```
Comment (benbovy) [inline xarray/core/formatting_html.py:254]: Do we want to collapse the indexes section by default, regardless of the number of items?

The list of names is redundant with the coordinates section, and it is probably fine if this section is one-click away (it's not primary information).

File: xarray/core/options.py [hunk 2]
```diff
@@ -169,34 +169,41 @@ class set_options:
...
         * ``True`` : to always expand data variables
         * ``False`` : to always collapse data variables
         * ``default`` : to expand unless over a pre-defined limit
+    display_expand_indexes : {"default", True, False}
+        Whether to expand the indexes section for display of
+        ``DataArray`` or ``Dataset``. Can be
+
+        * ``True`` : to always expand indexes
+        * ``False`` : to always collapse indexes
+        * ``default`` : to expand unless over a pre-defined limit
```
[the remark this answers is not in the document]
Comment (keewis) [reply xarray/core/options.py:206]: right, but I was planning to also use that in the text repr, and for that I don't think we should always collapse?

[the remark this answers is not in the document]
Comment (benbovy) [reply xarray/core/options.py:206]: Ah right, I forgot that it is also used for the text repr. I updated my suggestion above (or we can just leave as is).

**Comment by keewis** [comment]
> I added some commits (hope you don't mind) to fix the CSS

thanks a lot for that, @benbovy, I wouldn't have been able to figure that out on my own since my knowledge of CSS is pretty limited.

File: xarray/core/options.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (keewis) [reply xarray/core/options.py:206]: that addition makes sense to me!

File: xarray/core/formatting_html.py [hunk 3]
```diff
@@ -125,6 +125,40 @@ def summarize_vars(variables):
...
+
+    index_id = f"index-{uuid.uuid4()}"
+    preview = escape(inline_index_repr(index))
+    details = short_index_repr_html(index)
+
+    data_icon = _icon("icon-database")
+
+    return (
+        f"<div class='xr-index-name'><div>{name}</div></div>"
+        f"<div class='xr-index-preview xr-preview'>{preview}</div>"
```
Comment (benbovy) [inline xarray/core/formatting_html.py:146]: For slightly better contrast, you can remove the `xr-preview` class here

```suggestion
        f"<div class='xr-index-preview'>{preview}</div>"
```

and update CSS with 

```css
.xr-index-preview {
  grid-column: 2 / 5;
  color: var(--xr-font-color2);
}
```

**Comment by benbovy** [comment]
LGTM, that's awesome! It will be super handy for quick debugging and experimenting with custom indexes.

**Comment by benbovy** [comment]
🚀

**Pull request merged**